crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2713,6 +2725,25 @@ impl<'db> Class<'db> {
     }
 }
 
+#[salsa::interned]
+pub struct TypeAliasType<'db> {
+    #[return_ref]
+    pub name: ast::name::Name,
+
+    rhs_scope: ScopeId<'db>,
+}
+
+impl TypeAliasType<'_> {
+    pub fn value(self, db: &dyn Db) -> Type {
```

Comment:
This should be a `salsa::tracked` because it uses `scope.node` or we should document that this function should never be called from a query that's from another file than the current query.

```suggestion
#[salsa::tracked]
impl TypeAliasType<'_> {
    #[salsa::tracked]
    pub fn value(self, db: &dyn Db) -> Type {
```